endpoints/v2/manifest.py

```diff
@@ -109,14 +124,13 @@ def fetch_manifest_by_tagname(namespace_name, repo_name, manifest_ref):
 @v2_bp.route(MANIFEST_DIGEST_ROUTE, methods=["GET"])
 @disallow_for_account_recovery_mode
 @parse_repository_name()
-@process_registry_jwt_auth(scopes=["pull"])
+@process_registry_jwt_auth(scopes=FETCH_MANIFEST_SCOPES)
 @require_repo_read
 @anon_protect
 def fetch_manifest_by_digest(namespace_name, repo_name, manifest_ref):
     repository_ref = registry_model.lookup_repository(namespace_name, repo_name)
     if repository_ref is None:
-        image_pulls.labels("v2", "manifest", 404).inc()
-        raise NameUnknown()
+        _handle_repo_not_found(repo_name, FETCH_MANIFEST_SCOPES)
```

Comment:
Instead, I think this should be part of a PR that implements a pull-through proxy. The check should happen _after_ the auth header has been processed. i.e The permission system determined that the requester has the correct access. From there, if a repo doesn't exist, instead of just 404, do whatever extra work that is needed: check if the feature enabled, is the namespace setup as a pull-through proxy, ...

Reply:
this piece of code would never be reached because the `require_repo_read` decorator did a None check for the `repository_ref` and returned a 401. feel free to double check my tests though, as I aimed to maintain the same (weird if you ask me) behaviour.

Reply:
The fact that this condition was never reached is exactly the reason why I've changed the decorator - so that the endpoint would be reached and the pull through proxy code would be able to run as per your suggestion.

Reply:
It is though... Take the scenario where I'm correctly authenticated, but try something like `docker pull quay.io/my_valid_namespace/some-non-existent-repo`

Reply:
that's true! I hadn't been able to reach that code because I was using an anonymous user and mistakenly assumed the behaviour was the same.

Reply:
I've updated the PR now so that the behaviour would only change for anonymous users - but this is something we want to verify with product. I've asked on [slack](https://coreos.slack.com/archives/C02N3E07R9D/p1642168689010900). Let's wait and see what's the veredict.